src/Xamarin.Forms.EasyLoading.Multi/Platforms/Shared/LoadingLayout.cs

```diff
@@ -64,17 +79,48 @@ public static DataTemplateSelector GetLoadingTemplateSelector(BindableObject b)
             return (DataTemplateSelector)b.GetValue(LoadingTemplateSelectorProperty);
         }
 
+        public static void SetLoadingState(BindableObject b, LoadingState value)
+        {
+            b.SetValue(LoadingStateProperty, value);
+        }
+
+        public static LoadingState GetLoadingState(BindableObject b)
+        {
+            return (LoadingState)b.GetValue(LoadingStateProperty);
+        }
+
         static void OnIsLoadingChanged(BindableObject bindable, bool oldValue, bool newValue)
         {
             // Swap out the current children for the Loading Template.
             if (oldValue != newValue && newValue)
             {
+                SetLoadingState(bindable, LoadingState.Loading); // Backward Compat for IsLoadingProperty
                 GetLoadingLayoutController(bindable).SwitchToLoadingTemplate();
             }
             else if (oldValue != newValue && !newValue)
+            {
+                SetLoadingState(bindable, LoadingState.Success); // Backward Compat for IsLoadingProperty
+                GetLoadingLayoutController(bindable).SwitchToContent();
+            }
+        }
+
+        static void OnLoadingStateChanged(BindableObject bindable, LoadingState oldValue, LoadingState newValue)
```

Comment:
Wouldn't it be better to write it this way? So we avoid doing multiple comparisons and getting the loading layout controller more than one time.

```c#
static void OnLoadingStateChanged(BindableObject bindable, LoadingState oldValue, LoadingState newValue)
{
    if(oldValue == newValue)
        return;

    // Swap out the current children for the Loading Template.
    var loadingLayoutController = GetLoadingLayoutController(bindable);
    switch(GetLoadingState(bindable))
    {
        case LoadingState.Success:
            loadingLayoutController.SwitchToContent();
            break;
        case LoadingState.Loading:
            loadingLayoutController.LoadingStateTemplate = GetLoadingTemplate(bindable);
            loadingLayoutController.SwitchToLoadingTemplate();
            break;
        case LoadingState.Error:
            loadingLayoutController.LoadingStateTemplate = GetErrorTemplate(bindable);
            loadingLayoutController.SwitchToLoadingTemplate();
            break;
        default:
            throw new NotImplementedException($"LoadingState: {LoadingState}");
    }
}
```